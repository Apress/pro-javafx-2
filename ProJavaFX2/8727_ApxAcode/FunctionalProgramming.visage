/*
 * @test
 * @run
 */

{
  var addN: function(:Integer):function(:Integer):Integer;
  addN = function(n) {
    function(i:Integer) {
      i + n
    }
  };

  var addTen = addN(10);
  println("addTen(4) = {addTen(4)}.");
}
