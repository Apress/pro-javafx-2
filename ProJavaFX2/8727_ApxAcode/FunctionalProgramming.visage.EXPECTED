addTen(4) = 14.
